### PR TITLE
Feature/ibor 736

### PR DESCRIPTION
**Pull request by cavem-finbourne**
# Pull Request Checklist

- [x] Changes follow the [style guide](https://github.com/finbourne/sample-notebooks/blob/master/docs/FINBOURNE%20notebook%20style%20guide.ipynb)
- [x] Tests pass
- [x] Raised the PR against the `develop` branch
- [x] Notebook outputs do not contain any priviledged data

# Description of the PR
This PR aims to improve the robustness of Bond.ipynb as it has failed in the pipeline on the last two Monday mornings. It does this by making the notebook use transaction types in a custom scope rather than the transaction types in the default scope which are more likely to be modified by other notebooks/tests/users.   
Have also added deletion of the portfolio created by previous runs of the notebook so that, when the notebook tries to create the portfolio, it succeeds rather than failing because one already exists. This means that the portfolio used by the test will have the `transaction_type_scope `set to the custom scope.

Have run the notebook multiple times locally with success.

### CODE REVIEW

File: examples/use-cases/instruments/Bond.ipynb [hunk 1]
```diff
@@ -59,12 +59,13 @@
     "In this example, we will be using a 30Y US Treasury bond.\n",
```
[the remark this answers is not in the document]
Comment (cavem-finbourne) [reply examples/use-cases/instruments/Bond.ipynb:59]: Was aiming for uniqueness, I've taken it out

[the remark this answers is not in the document]
Comment (cavem-finbourne) [reply examples/use-cases/instruments/Bond.ipynb:59]: Added some commentary

[the remark this answers is not in the document]
Comment (cavem-finbourne) [reply examples/use-cases/instruments/Bond.ipynb:59]: Maybe I'm not clear on this.
If I add a fairly unique static string to the end of the code, there will still be concurrency risks, no?
I've changed it so it's randomly generating the code each time but then deleting it at the end of the nbk.

[the remark this answers is not in the document]
Comment (cavem-finbourne) [reply examples/use-cases/instruments/Bond.ipynb:59]: Hopefully fixed this now

Comment (rizwansaeed) [inline examples/use-cases/instruments/Bond.ipynb:59]: <p>just for extra safety i'd wrap this in another <code>try-except</code>	so it doesnt fail the nb if it fails (the delete failing isnt going to be the end of the world)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/739/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='51'/>

Comment (cavem-finbourne) [reply examples/use-cases/instruments/Bond.ipynb:59]: Wrapped the deletion in try-except

**Review by rizwansaeed — APPROVED** [review]

**Review by rizwansaeed — APPROVED** [review]

**Review by rizwansaeed — APPROVED** [review]

**Review by adamMQM — APPROVED** [review]

**Review by adamMQM — APPROVED** [review]

**Review by adamMQM — DISMISSED** [review]

**Review by rizwansaeed — APPROVED** [review]

**Pull request merged**